File: takeItEasy-domain/src/main/java/com/sudosoo/takeiteasy/controller/NoticeController.java
package com.sudosoo.takeiteasy.controller;

import com.sudosoo.takeiteasy.service.NoticeService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.servlet.mvc.method.annotation.SseEmitter;

@RestController
@RequestMapping("/api/notice")
@RequiredArgsConstructor
public class NoticeController {

    private final NoticeService noticeService;

    @GetMapping(value = "/subscribe", produces = "text/event-stream")
    public SseEmitter subscribe(@RequestParam String memberName,
                                @RequestHeader(value = "Last-Event-ID", required = false, defaultValue = "") String lastEventId) {
        return noticeService.subscribe(memberName, lastEventId);
    }


}

File: takeItEasy-presentation/src/main/kotlin/com/sudosoo/takeItEasy/controller/NoticeController.kt
package com.sudosoo.takeItEasy.controller

import com.sudosoo.takeItEasy.service.NoticeService
import com.sudosoo.takeiteasy.service.NoticeService
import org.springframework.web.bind.annotation.*
import org.springframework.web.servlet.mvc.method.annotation.SseEmitter

@RestController
@RequestMapping("/api/notice")
class NoticeController(val noticeService: NoticeService) {

    @GetMapping("/subscribe", produces = ["text/event-stream"])
    fun subscribe(
        @RequestParam memberName : String,
        @RequestHeader(value = "Last-Event-ID", required = false, defaultValue = "") lastEventId : String
    ): SseEmitter {
        return noticeService.subscribe(memberName, lastEventId)
    }
}